crawler/simple_crawler.py

```diff
@@ -14,11 +16,17 @@ def __init__(self, status_code, reason):
 
 
 def open_file(url):
-    status_code, reason = check_path(url)
-    response = Response(status_code, reason)
+    if check_if_html_cached(url):
+        response = Response(200, 'OK')
+        response.text = get_html_cache(url)
+    else:
+        status_code, reason = check_path(url)
+        response = Response(status_code, reason)
     if response.status_code == 200:
-        with open(url, 'rb') as f:
-            response.text = f.read()
+        if response.text is None:
+            with open(url, 'rb') as f:
```

Comment:
doesn´t `rb` mean read in binary mode?

Aah, while HTML files are text, images are binary.  But, I _think_ we only ever copy binary files - I don´t think we open them.

Reply:
Yep, let me change it to 'r'